RFC-0036-hardware-accelerators-pytorch.org.md

```diff
@@ -0,0 +1,89 @@
+# Hardware accelerators at pytorch.org
+
+**Authors:**
+* @bsochack
+* @sujoysaraswati
+* @fzhao3
+* @jgong5
+
+
+## **Summary**
+The main page of pytorch.org only shows PyTorch Get-Started  on CUDA, ROCm and CPU, but the hardware accelerator support is more diverse and the ecosystem is rapidly evolving with new accelerators like Intel Gaudi, Google TPU, etc.
+The proposal in this RFC is to enable the visibility of broad set of accelerators that supports PyTorch for the community, and create a space for hardware vendors to provide instructions and details about their PyTorch support.  
+
+
+## **Motivation**
+There are 2 primary motivations behind this RFC:
+1. Evangelize PyTorch as an open source framework that supports a wide variety of hardware.
+   * PyTorch is supported on more compute platforms than CUDA, ROCm and CPU .
+   * PyTorch can maintain and enhance its competitive position in the machine learning frameworks  by enabling the visibility of rich set of hardware accelerators on which PyTorch can run.
+2. Provide a consolidated entry point where to look for more information about supported hardware accelerators.
+   * pytorch.org is a starting point for many users. The page can help users to find more details.
+
+
+
+## **Proposed Implementation**
+The proposal is to enhance pytorch.org by adding 3  new sections:
+* Add new compute platforms to “Install PyTorch” section at the main page of pytorch.org.
+* Add new compute platforms to https://pytorch.org/get-started/.
+* Add new “Compute Platforms” section at the main page of pytorch.org.
+
+### PyTorch integration types
+There are at least 2 ways how compute platforms are integrated with PyTorch:
+1. In-tree – CPU, CUDA, ROCm are developed, built and tested in PyTorch environment. PyTorch is ensuring quality criteria. This approach is limited to only a few compute platforms and it does not scale with number of compute platforms.
+2. Out-of-tree – Integration of other compute platforms like Intel Gaudi is done via additional python package (extension) that needs to be installed on top of PyTorch CPU package. Development, built and testing is done outside of PyTorch. In this case:
```

Comment:
Does "out of tree" hardware accelerator refer to an accelerator that implements logic outside of PyTorch upstream, including devices accessed via privateuse1?

If so, could we specify "out-of-tree (including privateuse keys)" here?

Reply:
Yes, "out of tree" refers to "integration logic" that is implemented outside of PyTorch upstream. It is applicable to privateuse1 and other accelerators with device specific dispatch key (i.e. Gaudi uses HPU key)